src/metabase/models/collection.clj

```diff
@@ -382,7 +386,12 @@
   (let [parent-id           (or (:id parent-collection) "")
         child-literal       (if (collection.root/is-root-collection? parent-collection)
                               "/"
-                              (format "%%/%s/" (str parent-id)))]
+                              (format "%%/%s/" (str parent-id)))
+        collection-ids      (if (= collection-ids :all)
+                              (t2/select-pks-set :model/Collection :archived (or
```

Comment:
There was an existing bug here. When looking for collection children, we didn't care at all about `:archived`. So with collections like `/A/B/C` where B is archived, we'd show A as having no effective children.

I'm a little concerned about performance here and still thinking about whether we can help that.